napari/_qt/_qapp_model/_tests/test_view_menu.py

```diff
@@ -49,22 +49,28 @@ def test_toggle_axes_scale_bar_attr(
 
 
 @skip_local_popups
-def test_toggle_fullscreen(make_napari_viewer):
+def test_toggle_fullscreen(make_napari_viewer, qtbot):
     """Test toggle fullscreen action."""
     action_id = 'napari.window.view.toggle_fullscreen'
     app = get_app()
     viewer = make_napari_viewer(show=True)
 
     # Check initial default state (no fullscreen)
-    assert not viewer.window._qt_window._fullscreen_flag
+    assert not viewer.window._qt_window.isFullScreen()
 
     # Check fullscreen state change
     app.commands.execute_command(action_id)
-    assert viewer.window._qt_window._fullscreen_flag
+    if sys.platform == 'darwin':
+        # On macOS, wait for the animation to complete
+        qtbot.wait(250)
```

Comment:
Gave this a check and makes sense to me :+1: I think the main element that causes a need for a little wait in macOS is that going into full screen mode there makes the app move to is own dedicated desktop which is an operation that is not instant and involves some sort of OS level animation indeed.

For more info related with the considerations when using `QWidget.showFullScreen` you can check https://doc.qt.io/qt-6/qwidget.html#showFullScreen